carbonmark-api/src/routes/purchases/[id]/get.ts

```diff
@@ -29,14 +37,42 @@ const handler = async (
   const [standard, registryProjectId] = CreditId.splitProjectId(
     purchase.listing.project.key
   );
-  const project = await fetchCarbonProject(sdk, {
-    registry: standard,
-    registryProjectId,
-  });
+  let fetchCarbonProjectMethod: FetchCarbonProjectMethod;
+  let fetchCarbonProjectArgs: FetchCarbonProjectArgs;
+
+  switch (standard) {
+    case REGISTRIES["ICR"].id: {
+      const { ICR_API_URL } = ICR_API(network);
+      fetchCarbonProjectMethod = ICR_API_URL;
+      fetchCarbonProjectArgs = {
+        // @todo use tokenAddress in place of the serialization for purchases as the serialization is not available on marketplace subgraph
+        contractAddress: purchase.listing.tokenAddress,
+        vintage: purchase.listing.project.vintage,
+        network: network,
+      };
+      break;
+    }
+    default:
+      fetchCarbonProjectMethod = sdk;
+      fetchCarbonProjectArgs = {
+        registry: standard,
+        registryProjectId,
+        network: network,
+      };
+      break;
+  }
+  const project = await fetchCarbonProject(
+    fetchCarbonProjectMethod,
+    fetchCarbonProjectArgs
+  );
```

Comment:
You can put the algorithm that computes the `fetchCarbonProjectMethod` and `fetchCarbonProjectArgs` inside the `fetchCarbonProject`. This code is dupliccated in the `/projects/:id` route

Reply:
refactored project data to cms so this is now removed